js/bufferResume.js

```diff
@@ -0,0 +1,56 @@
+
+/*
+ * This file contains the service used to record the last 
+ * accessed buffer and return to it when reconnecting to 
+ * the relay.
+ */
+(function() {
+'use strict';
+
+var bufferResume = angular.module('bufferResume', []);
+
+bufferResume.service('bufferResume', ['settings', function(settings){
+    var resumer = {};
+    var key = settings.host + ":" + settings.port;
+
+    // Hold the status that we were able to find the previously accessed buffer
+    //   and reload it.  If we cannot, we'll need to know so we can load the default
+    var hasResumed = false;
+
+    // Store the current buffer as having been accessed.  We can later retrieve it and compare 
+    //   we recieve info from weechat to determine if we should switch to it.
+    resumer.record = function(activeBuffer){
+        var subSetting = settings.currentlyViewedBuffers;
+        subSetting[key] = activeBuffer.id;
+        settings.currentlyViewedBuffers = subSetting;
+    };
+
+    // See if the requested buffer information matches the last recorded access.  If so, 
+    //   the handler should switch to this buffer.
+    resumer.shouldResume = function(buffer){
+        var savedBuffer = settings.currentlyViewedBuffers[key];
+        if(savedBuffer === 'undefined' || savedBuffer === undefined){return false;}
+
+        if(!hasResumed){
```

Comment:
style - we use spaces after `if` and between `)` and `{` in the rest of the project. This line, and the one above/below it.